src/unitxt/inference.py

```diff
@@ -47,6 +47,9 @@ class TextGenerationInferenceOutput:
 class InferenceEngine(abc.ABC, Artifact):
     """Abstract base class for inference."""
 
+    label: str
```

Comment:
I'm not sure we want to face having a model_name. Having an optional label is fine, but make it optional. What is the need for it?